pkgs/applications/science/math/scilab-bin/default.nix

```diff
@@ -1,100 +1,119 @@
-{ stdenv, fetchurl, lib, xorg }:
+{ lib, stdenv, fetchurl, undmg, makeWrapper, xorg }:
 
 let
-  badArch = throw "scilab-bin requires i686-linux or x86_64-linux";
-
-  architecture =
-    if stdenv.hostPlatform.system == "i686-linux" then
-      "i686"
-    else if stdenv.hostPlatform.system == "x86_64-linux" then
-      "x86_64"
-    else
-      badArch;
-in
-stdenv.mkDerivation rec {
+  badArch = throw "scilab-bin requires darwin or x86_64-linux";
+
   pname = "scilab-bin";
   version = "6.1.1";
 
   src = fetchurl {
-    url = "https://www.scilab.org/download/${version}/scilab-${version}.bin.linux-${architecture}.tar.gz";
+    url =
+      if stdenv.hostPlatform.system == "aarch64-darwin" then
+        "https://www.utc.fr/~mottelet/scilab/download/${version}/scilab-${version}-accelerate-arm64.dmg"
+      else if stdenv.hostPlatform.system == "x86_64-darwin" then
+        "https://www.utc.fr/~mottelet/scilab/download/${version}/scilab-${version}-x86_64.dmg"
+      else if stdenv.hostPlatform.system == "x86_64-linux" then
+        "https://www.scilab.org/download/${version}/scilab-${version}.bin.linux-x86_64.tar.gz"
```

Comment:
Can we fetch the donwload from one domain? Also please use an attr to select the download instead of an if else chain.

Reply:
The macOS builds are taken from the link given on the [official website](https://www.scilab.org/download/scilab-6.1.1), which points to another domain, so the answer is no, unfortunately.